google-cloud-tools-plugin/src/com/google/gct/idea/debugger/CloudDebugGlobalPoller.java

```diff
@@ -166,13 +170,27 @@ private void fireBreakpointsChanged(@NotNull CloudDebugProcessState state) {
    *
    * @param state represents the target debuggee to query
    */
-  private void pollForChanges(@NotNull final CloudDebugProcessState state) {
+  void pollForChanges(@NotNull final CloudDebugProcessState state) {
     final Debugger client = CloudDebuggerClient.getShortTimeoutClient(state);
     if (client == null) {
-      // It is ok if there is no client.  We may want to consider notifying the user that
-      // background polling is disabled until they login, but for now, we can just doc it.
-      LOG.info("no client available attempting to checkForChanges");
-      return;
+      if (state.isListenInBackground()) {
```

Comment:
This PR is for the case when the user explicitly logs out. In this case the locally cached user credentials are deleted during logout. When the next time getShortTimeoutClient() is called, it cannot find the credentials and therefore does not attempt a server request, but returns a `null` instead of a Debugger instance.
We can change the return type of the method to be a (status,Debugger) tuple and the status could show the error reason if Debugger is null.